### PR TITLE
EVMTool improvements

### PR DESCRIPTION
**Pull request by shemnon**
* Support clique genesis (needs a nodekey)
* Write genesis allocations to world state
* Report total gas (EVM operations + intrinsic cost of TX)

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

### CODE REVIEW

File: ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/CliqueGenesisFileModule.java [hunk 1]
```diff
@@ -18,13 +18,18 @@
...
+import org.hyperledger.besu.crypto.NodeKey;
+import org.hyperledger.besu.crypto.SECP256K1;
 import org.hyperledger.besu.ethereum.core.BlockHeaderFunctions;
 import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
 
 import javax.inject.Named;
 
 class CliqueGenesisFileModule extends GenesisFileModule {
 
+  NodeKey nodeKey = new NodeKey(new KeyPairSecurityModule(SECP256K1.KeyPair.generate()));
```
Comment (matkt) [inline ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/CliqueGenesisFileModule.java:31]: In my opinion I think it is possible to add `private final` here

File: ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/EvmToolCommand.java [hunk 2]
```diff
@@ -232,13 +234,27 @@ public void run() {
...
+                  Wei.ZERO,
+                  null,
+                  callData,
+                  sender,
+                  Optional.empty());
+
+          final Gas intrinsicGasCost =
+              component.getGasCalculatorAtBlock().apply(0).transactionIntrinsicGasCost(tx);
+          Gas evmGas = gas.minus(messageFrame.getRemainingGas());
           System.out.println(
```
Comment (matkt) [inline ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/EvmToolCommand.java:252]: Why don't we use the logger present in the class to log this? `LogManager.getLogger()`

Comment (shemnon) [reply ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/EvmToolCommand.java]: Logger create too much extra line junk with the default setup for what should be a CLI focused tool.  However, I did change to use the printstream from the PicoCLI invocation.

File: ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/CliqueGenesisFileModule.java (hunk 1 above)
Comment (shemnon) [reply ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/CliqueGenesisFileModule.java]: fixed.

**Pull request merged**